influxql/functions.go

```diff
@@ -255,6 +266,125 @@ func ReduceMean(values []interface{}) interface{} {
 	return nil
 }
 
+// ReduceMedian computes the median of values
+func ReduceMedian(values []interface{}) interface{} {
+	var data []float64
+	// Collect all the data points
+	for _, value := range values {
+		if value == nil {
+			continue
+		}
+		data = append(data, value.([]float64)...)
+	}
+
+	length := len(data)
+	if length < 2 {
+		if length == 0 {
+			return nil
+		}
+		return data[0]
+	}
+	middle := length / 2
+	var sortedRange []float64
+	if length%2 == 0 {
+		sortedRange = getSortedRange(data, middle-1, 2)
+		var low, high = sortedRange[0], sortedRange[1]
+		return low + (high-low)/2
+	} else {
+		sortedRange = getSortedRange(data, middle, 1)
+		return sortedRange[0]
+	}
+}
+
+func getSortedRange(data []float64, start int, count int) []float64 {
```

Comment:
what's your convention/requirements? i'll try to add something if you like.

i just tried to make the names descriptive since i think that things like

```go
// MapMin collects the values to pass to the reducer
```
are kind of redundant.